File: spring-data-requery/src/main/java/org/springframework/data/requery/mapping/RequeryPersistentProperty.java
package org.springframework.data.requery.mapping;

import org.springframework.data.mapping.PersistentProperty;

/**
 * org.springframework.data.requery.mapping.RequeryPersistentProperty
 *
 * @author debop
 * @since 18. 6. 7
 */
public interface RequeryPersistentProperty extends PersistentProperty<RequeryPersistentProperty> {

    boolean isIdProperty();

    boolean isTransient();

    boolean isAssociation();

    boolean isEmbedded();

    boolean hasIndex();

    String getFieldName();

}

File: spring-data-requery/src/main/kotlin/org/springframework/data/requery/mapping/RequeryPersistentProperty.kt
package org.springframework.data.requery.mapping

import org.springframework.data.mapping.PersistentProperty

/**
 * [PersistentProperty] for Requery
 *
 * @author debop
 */
interface RequeryPersistentProperty: PersistentProperty<RequeryPersistentProperty> {

    override fun isIdProperty(): Boolean

    override fun isTransient(): Boolean

    override fun isAssociation(): Boolean

    fun isEmbedded(): Boolean

    fun hasIndex(): Boolean

    fun getFieldName(): String
}